### PR TITLE
IMAP - Make reactive SetACLProcessor/DeleteACLProcessor

### PR DESCRIPTION
**Pull request by vttranlina**

### CODE REVIEW

File: protocols/imap/src/main/java/org/apache/james/imap/processor/DeleteACLProcessor.java [hunk 1]
```diff
@@ -59,79 +62,64 @@ public DeleteACLProcessor(MailboxManager mailboxManager, StatusResponseFactory f
...
+                error -> Mono.fromRunnable(() -> taggedBad(request, responder,
+                    new HumanReadableText(
+                        HumanReadableText.UNSUPPORTED_RIGHT_KEY,
+                        HumanReadableText.UNSUPPORTED_RIGHT_DEFAULT_VALUE,
+                        error.getUnsupportedRight()))))
+            .onErrorResume(MailboxNotFoundException.class,
+                error -> Mono.fromRunnable(() -> no(request, responder, HumanReadableText.MAILBOX_NOT_FOUND)))
+            .onErrorResume(MailboxException.class,
+                error -> {
+                    LOGGER.error("{} failed for mailbox {}", request.getCommand().getName(), mailboxName, error);
```
Comment (chibenwa) [inline protocols/imap/src/main/java/org/apache/james/imap/processor/DeleteACLProcessor.java:89]: Please manage to get contextualized logd

File: protocols/imap/src/main/java/org/apache/james/imap/processor/SetACLProcessor.java [hunk 2]
```diff
@@ -62,19 +67,41 @@ public SetACLProcessor(MailboxManager mailboxManager, StatusResponseFactory fact
...
+                error -> Mono.fromRunnable(() -> taggedBad(request, responder,
+                    new HumanReadableText(
+                        HumanReadableText.UNSUPPORTED_RIGHT_KEY,
+                        HumanReadableText.UNSUPPORTED_RIGHT_DEFAULT_VALUE,
+                        error.getUnsupportedRight()))))
+            .onErrorResume(MailboxNotFoundException.class,
+                error -> Mono.fromRunnable(() -> no(request, responder, HumanReadableText.MAILBOX_NOT_FOUND)))
+            .onErrorResume(MailboxException.class,
+                error -> {
+                    LOGGER.error("{} failed for mailbox {}", request.getCommand().getName(), mailboxName, error);
```
Comment (chibenwa) [inline protocols/imap/src/main/java/org/apache/james/imap/processor/SetACLProcessor.java:94]: Idem applymdc context upon logs

File: protocols/imap/src/main/java/org/apache/james/imap/processor/DeleteACLProcessor.java (hunk 1 above)
Comment (chibenwa) [reply protocols/imap/src/main/java/org/apache/james/imap/processor/DeleteACLProcessor.java:89]: Use MDC context propagation in reactive stack (context)

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by Arsnael** [comment]
```
02:34:28,434 [INFO] There is 1 error reported by Checkstyle 10.3.4 with checkstyle.xml ruleset.
02:34:28,438 [ERROR] src/main/java/org/apache/james/mailbox/RightManager.java:[156,5] (whitespace) EmptyLineSeparator: 'METHOD_DEF' should be separated from previous line.
```

**Review by quantranhong1999 — APPROVED** [review]

**Comment by Arsnael** [comment]
https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1505/3/testReport/

I think related

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreRightManager.java [hunk 3]
```diff
@@ -169,19 +169,23 @@ public MailboxACL listRights(MailboxId mailboxId, MailboxSession session) throws
...
-            .doOnNext(Throwing.consumer(mapper -> assertSharesBelongsToUserDomain(mailboxPath.getUser(), mailboxACLCommand)))
+            .doOnNext(Throwing.consumer(mapper -> assertSharesBelongsToUserDomain(mailboxPath.getUser(), mailboxACLCommand)).sneakyThrow())
             .flatMap(mapper -> mapper.findMailboxByPath(mailboxPath)
-                .doOnNext(Throwing.consumer(mailbox -> assertHaveAccessTo(mailbox, session)))
+                .doOnNext(Throwing.<Mailbox>consumer(mailbox -> assertHaveAccessTo(mailbox, session)).sneakyThrow())
                 .flatMap(mailbox -> mapper.updateACL(mailbox, mailboxACLCommand)
-                    .flatMap(aclDiff -> dispatchACLUpdateEvent(session, mailbox, aclDiff))));
+                    .flatMap(aclDiff -> dispatchACLUpdateEvent(session, mailbox, aclDiff))))
+            .onErrorResume(e -> {
+                System.out.println(e);
```
Comment (chibenwa) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreRightManager.java:186]: Oups

**Comment by Arsnael** [comment]
https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1505/4/testReport/

I think those are related too :)

**Review by Arsnael — APPROVED** [review]

File: mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java [hunk 4]
```diff
@@ -2933,7 +2933,7 @@ void hasRightShouldThrowOnUnknownMailbox() {
                     MailboxPath.forUser(USER_1, "notFound"),
                     MailboxACL.Right.Administer,
                     session))
-                .isInstanceOf(MailboxNotFoundException.class);
+                .hasRootCauseExactlyInstanceOf(MailboxNotFoundException.class);
```
Comment (chibenwa) [inline mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java:2936]: :-1: we should manage to unwrap the exception in order not to break things.

CF MailboxReactorUtils for blocking.

**Review by chibenwa — APPROVED** [review]

**Comment by vttranlina** [comment]
Ci error: 
```
13:19:56.164 [ERROR] o.a.j.c.ClamAVScan - Exception caught acquiring main socket to CLAMD on localhost/0:0:0:0:0:0:0:1 on port 33920: Connection refused (Connection refused)
```
It looks not relate to this pr, 
I tried to run on locally and it succeeded `james-server-task-distributed`

Let me rebase master

**Pull request merged**